pkg/controller/nodeipam/ipam/cloud_cidr_allocator.go

```diff
@@ -164,10 +164,11 @@ func (ca *cloudCIDRAllocator) worker(stopChan <-chan struct{}) {
 				klog.Errorf("Error updating CIDR for %q: %v", workItem, err)
 				if canRetry, timeout := ca.retryParams(workItem); canRetry {
 					klog.V(2).Infof("Retrying update for %q after %v", workItem, timeout)
-					time.AfterFunc(timeout, func() {
+					timer := time.AfterFunc(timeout, func() {
 						// Requeue the failed node for update again.
 						ca.nodeUpdateChannel <- workItem
 					})
+					timer.Stop()
```

Comment:
This brakes current, correct behavior as you will cancel the timer/function just after creation so the callback will never be called.

Did you test this?

Reply:
Shouldn't time.AfterFunc be stopped after execution？

Reply:
Yes，timer.Stop() should not be on line 171, perhaps he should let the GC collect itself，You are right

Reply:
Yeah, I think `time.AfterFunc` and `time.After` and designed to be collected by the GC and there is nothing we (developers) need to do.

Reply:
time.AfterFunc can sometimes be recycled manually and may need to be done depending on the scenario, but this is not the place to do it，I'm going to close this PR. thank you